server/src/bin/ingestion-worker.rs

```diff
@@ -1019,6 +1019,18 @@ pub async fn readd_error_to_queue(
                 log::error!("Failed to create event: {:?}", err);
             });
 
+            let mut redis_conn = redis_pool
+                .get()
+                .await
+                .map_err(|err| ServiceError::BadRequest(err.to_string()))?;
+
+            redis::cmd("lpush")
+                .arg("dead_letters")
+                .arg(old_payload_message)
+                .query_async(&mut *redis_conn)
+                .await
+                .map_err(|err| ServiceError::BadRequest(err.to_string()))?;
+
```

Comment:
We should also remove the item from the processing queue here, to keep the processing queue for only what is currently being processed